builtins: implement ST_S2Covering

otan opened the pull request:
Planning to add non-default index configurations for this in a later PR.

Release note (sql change): Implement ST_S2Covering, which returns the S2
coverings used for indexing for geometry and geography types.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/54278)
<!-- Reviewable:end -->

otan committed:
builtins: implement ST_S2Covering

Planning to add non-default index configurations for this in a later PR.

Release note (sql change): Implement ST_S2Covering, which returns the S2
coverings used for indexing for geometry and geography types.

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 6 of 6 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/54278#-:-MHC9REW2uCRZZWClBxc:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[docs/generated/sql/functions.md, line 2009 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/54278#-MHC4x1o1I5mo96zAyWs:-MHC4x1p2KrosuJP0_zT:bmqu3i7) ([raw file](https://github.com/cockroachdb/cockroach/blob/7bb231a1aa8a078c1863e6b69eadc99ad33aba17/docs/generated/sql/functions.md#L2009)):*
> ```Markdown
> <tr><td><a name="st_rotate"></a><code>st_rotate(g: geometry, angle_radians: <a href="float.html">float</a>, origin_x: <a href="float.html">float</a>, origin_y: <a href="float.html">float</a>) &rarr; geometry</code></td><td><span class="funcdesc"><p>Returns a modified Geometry whose coordinates are rotated around the provided origin by a rotation angle.</p>
> </span></td></tr>
> <tr><td><a name="st_s2covering"></a><code>st_s2covering(geography: geography) &rarr; geography</code></td><td><span class="funcdesc"><p>Returns a geography which represents the S2 covering used by the index using the default index configuration.</p>
> ```

is the "spatial type" st_ prefix only for SQL/MM standard or is it ok to use it for anything spatial related?


<!-- Sent from Reviewable.io -->

otan commented:
___
*[docs/generated/sql/functions.md, line 2009 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/54278#-MHC4x1o1I5mo96zAyWs:-MHCAwN57GaAg04GCAUr:bild7jl) ([raw file](https://github.com/cockroachdb/cockroach/blob/7bb231a1aa8a078c1863e6b69eadc99ad33aba17/docs/generated/sql/functions.md#L2009)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

is the "spatial type" st_ prefix only for SQL/MM standard or is it ok to use it for anything spatial related?
</blockquote></details>

i think we can just use it for anything spatial related. PostGIS does the same (mostly) :)


<!-- Sent from Reviewable.io -->
